src/actions/public/L1/readFinalizedWithdrawals.ts

```diff
@@ -0,0 +1,43 @@
+import { optimismPortalABI } from '@eth-optimism/contracts-ts'
+import type { Chain, PublicClient, Transport } from 'viem'
+import type { MessagePassedEvent } from '../../../index.js'
+import type { GetL2Chain, L1ActionBaseType } from '../../../types/l1Actions.js'
+import { OpStackL1Contract } from '../../../types/opStackContracts.js'
+import { readOpStackL1, type ReadOpStackL1Parameters } from './readOpStackL1.js'
+
+const ABI = optimismPortalABI
+const CONTRACT = OpStackL1Contract.OptimismPortal
+
+export type ReadFinalizedWithdrawalsParameters<
+  TChain extends Chain | undefined = Chain,
+> =
+  & { withdrawalHash: MessagePassedEvent['withdrawalHash'] }
+  & L1ActionBaseType<
+    GetL2Chain<TChain>,
+    typeof CONTRACT
+  >
+
+export async function readFinalizedWithdrawals<TChain extends Chain | undefined>(
+  client: PublicClient<Transport, TChain>,
+  {
+    withdrawalHash,
+    optimismPortalAddress,
+    l2Chain,
+  }: ReadFinalizedWithdrawalsParameters<TChain>,
+): Promise<boolean> {
+  const finalizedWithdrawal = await readOpStackL1(client, {
+    contract: CONTRACT,
+    abi: ABI,
+    functionName: 'finalizedWithdrawals',
+    l2Chain,
+    address: optimismPortalAddress,
+    args: [withdrawalHash],
+    chain: client.chain,
+  } as ReadOpStackL1Parameters<TChain, typeof ABI, 'finalizedWithdrawals'>)
+
+  if (finalizedWithdrawal === false) {
+    throw new Error(`Withdrawal with hash ${withdrawalHash} is not finalized`)
```

Comment:
Thanks for the thoroughness, but since this just returns false, I don't think we need to error. False is clear enough. With provenWithdrawals we wanted to avoid confusion with returning the struct with 0 values